tutorials/roofit/rf709_BarlowBeeston.py

```diff
@@ -0,0 +1,216 @@
+
+Asig2 = ROOT.RooRealVar("Asig", "Asig", 1, 0.01, 5000)
+Abkg2 = ROOT.RooRealVar("Abkg", "Abkg", 1, 0.01, 5000)
+
+# As before, construct the sum of signal2 and background2
+model2_tmp = ROOT.RooRealSumPdf(
+    "sp_ph", "sp_ph", ROOT.RooArgList(p_ph_sig2, p_ph_bkg2), ROOT.RooArgList(Asig2, Abkg2), True
+)
+
+# Construct the subsidiary poisson measurements constraining the statistical fluctuations
+hc_sigbkg = ROOT.RooHistConstraint("hc_sigbkg", "hc_sigbkg", ROOT.RooArgSet(p_ph_sig2, p_ph_bkg2))
+
+# Construct the joint model
+model2 = ROOT.RooProdPdf("model2", "model2", hc_sigbkg, Conditional=(model2_tmp, x))
+
+
+#  Fit all models to data and plot
+
+result0 = model0.fitTo(sumData, PrintLevel=0, Save=True)
+result1 = model1.fitTo(sumData, PrintLevel=0, Save=True)
+result2 = model2.fitTo(sumData, PrintLevel=0, Save=True)
+
+
+can = ROOT.TCanvas("can", "", 1500, 600)
+can.Divide(3, 1)
+
+pt = ROOT.TPaveText(-19.5, 1, -2, 25)
+pt.SetFillStyle(0)
+pt.SetBorderSize(0)
+
+
+can.cd(1)
+frame = x.frame(Title="No template uncertainties")
+# Plot data to enable automatic determination of model0 normalisation:
+sumData.plotOn(frame)
+model0.plotOn(frame, LineColor="b", VisualizeError=result0)
+# Plot data again to show it on top of model0 error bands:
+sumData.plotOn(frame)
+# Plot model components
+model0.plotOn(frame, LineColor="b")
+model0.paramOn(frame)
+
+sigData.plotOn(frame, MarkerColor="b")
+frame.Draw()
+
+pt_text1 = [
+    "No template uncertainties",
+    "are taken into account.",
+    "This leads to low errors",
+    "for the parameters A, since",
+    "the only source of errors",
+    "are the data statistics.",
+]
+for text in pt_text1:
+    pt.AddText(text)
+
+pt.DrawClone()
+
+
+can.cd(2)
+frame = x.frame(Title="Barlow Beeston for Sig & Bkg separately")
+sumData.plotOn(frame)
+model1.plotOn(frame, LineColor="b", VisualizeError=result1)
+# Plot data again to show it on top of error bands:
+sumData.plotOn(frame)
+model1.plotOn(frame, LineColor="b")
+model1.plotOn(frame, Components=p_ph_sig1, LineColor="kAzure")
+model1.plotOn(frame, Components=p_ph_bkg1, LineColor="r")
+model1.paramOn(frame, Parameters=ROOT.RooArgSet(Asig1, Abkg1))
+
+sigData.plotOn(frame, MarkerColor="b")
+frame.Draw()
+
+pt.Clear()
+
+pt_text2 = [
+    "With gamma parameters, the",
+    "signal & background templates",
+    "can adapt to the data.",
+    "Note how the blue signal",
+    "template changes its shape.",
+    "This leads to higher errors",
+    "of the scale parameters A.",
+]
+
+for text in pt_text2:
+    pt.AddText(text)
+
+pt.DrawClone()
+
+can.cd(3)
+frame = x.frame(Title="Barlow Beeston light for (Sig+Bkg)")
+sumData.plotOn(frame)
+model2.plotOn(frame, LineColor="b", VisualizeError=result2)
+# Plot data again to show it on top of model0 error bands:
+sumData.plotOn(frame)
+model2.plotOn(frame, LineColor="b")
+model2.plotOn(frame, Components=p_ph_sig2, LineColor="kAzure")
+model2.plotOn(frame, Components=p_ph_bkg2, LineColor="r")
```

Comment:
Before the ownership issues are solved in the backend, you have to define the components set explicitly and give it a name:
```Python
p_ph_sig2_set = ROOT.RooArgSet(p_ph_sig2)
p_ph_bkg2_set = ROOT.RooArgSet(p_ph_bkg2)
model2.plotOn(frame, Components=p_ph_sig2_set, LineColor="kAzure")
model2.plotOn(frame, Components=p_ph_bkg2_set, LineColor="r")
```